test/test_sched.rs

```diff
@@ -0,0 +1,36 @@
+
+#[cfg(any(target_os = "android",
+          target_os = "linux"))]
+#[test]
+fn test_sched_affinity() {
+    use nix::sched::{sched_getaffinity, sched_setaffinity, CpuSet};
+    use nix::unistd::Pid;
+    use std::mem;
+
+    // If pid is zero, then the mask of the calling process is returned.
+    let initial_affinity = sched_getaffinity(Pid::from_raw(0)).unwrap();
+    let mut at_least_one_cpu = false;
+    let mut last_valid_cpu = 0;
+    for field in 0..(8 * mem::size_of::<libc::cpu_set_t>()) {
```

Comment:
What does the magic number 8 mean?  You should remove it if possible, or define it as a const if not.

Reply:
This value comes from the `CpuSet::set()` and `CpuSet::is_set()` methods.
https://github.com/nix-rust/nix/blob/master/src/sched.rs#L63
```rust
        pub fn is_set(&self, field: usize) -> Result<bool> {
            if field >= 8 * mem::size_of::<libc::cpu_set_t>() {
                Err(Error::Sys(Errno::EINVAL))
            } else {
                Ok(unsafe { libc::CPU_ISSET(field, &self.cpu_set) })
            }
        }
```

That's because there is no easy way to know the size of a CpuSet and in the test I want to check each bits of the CpuSet.

I can do differently:
- add a method `CpuSet::size()` which will return this value.
- call `CpuSet::is_set()` until I get a `Errno::EINVAL`
What do you prefer ?

Reply:
Better to add a method to `CpuSet`.  But I would call it `count`, because that's what the C macros call it.